plugin/keyring_vault/vault_credentials_parser.cc

```diff
@@ -0,0 +1,127 @@
+#include <my_global.h>
+#include "vault_credentials_parser.h"
+#include <fstream>
+#include <algorithm>
+#include <iostream>
+
+namespace keyring
+{
+  struct Is_not_space
+  {
+    my_bool operator()(char c)
+    {
+      return !std::isspace(c);
+    }
+  };
+
+  static inline Secure_string* ltrim(Secure_string *s) {
```

Comment:
From what I have found on this topic, references should be used only for const parameters. From Google C++ standard:
"All parameters passed by reference must be labeled const."
From Stroustrup book:
[1] Use pass-by-value for small objects.
[2] Use pass-by-const-reference to pass large values that you don’t need to modify.
[3] Return a result as a return value rather than modifying an object through an argument.
[4] Use rvalue references to implement move (§3.3.2, §17.5.2) and forwarding (§23.5.2.1).
[5] Pass a pointer if ‘‘no object’’ is a valid alternative (and represent ‘‘no object’’ by nullptr).
[6] Use pass-by-reference only if you have to.
Stroustrup suggest pointer to mark nullptr possibility, but also says thatpass-by-reference it the last resort. I will change parameters to const reference whenever I can and we can take this discussion further then.